### PR TITLE
refactor: consolidate duplicate test methods

### LINKED ISSUE #1073: Merge duplicate definitions with testutil
Subtask for #832 

Found duplicated function definitions for `generateRandomlySizedTransactions` and `generateRandomlySizedBlobs`
Example ref:
`generateRandomlySizedTransaction(s)`
https://github.com/celestiaorg/celestia-app/blob/8b3dacb24c690fa650a7654f3a5a2f93b44a1c1d/pkg/shares/shares_test.go#L109

https://github.com/celestiaorg/celestia-app/blob/8b3dacb24c690fa650a7654f3a5a2f93b44a1c1d/pkg/shares/shares_test.go#L121

https://github.com/celestiaorg/celestia-app/blob/8b3dacb24c690fa650a7654f3a5a2f93b44a1c1d/pkg/prove/proof_test.go#L189

https://github.com/celestiaorg/celestia-app/blob/8b3dacb24c690fa650a7654f3a5a2f93b44a1c1d/pkg/prove/proof_test.go#L201

Identified Duplicates:
| Duplicates                                                     | Packages         |
|----------------------------------------------------------------|------------------|
| `generateRandomlySizedTransactions`,`generateRandomlySizedTxs` | `shares`,`prove` |
| `generateRandomTransaction`, `generateRandomTxs`               | `shares`,`prove` |
| `generateRandomlySizedBlobs`                                   | `shares`,`prove` |
| `generateRandomBlob`                                           | `shares`,`prove` |
|`GenerateKeyringSigner`, `generateKeyringSigner`, `generateKeyring` | `app`, `app_test`, `testutil`, `x/blob/types` |
|`GenerateManyRawWirePFB`, `GenerateManyRawSendTxs`, `generateRawTx`, `generateManyRawWirePFB`, `generateManyRawSendTxs` | `app`, `app_test`, `blobtestutil` |

Proposal:
- Create a `factory` package inside `testutil` and move all these generators to factory package

### PR DESCRIPTION
**Pull request by vidhanarya**
## Overview
Closes #1073

Affected packages
1. `pkg` -> `shares` & `prove`
    - Remove duplicated `generateRandomTransaction`, `generateRandomlySizedTransactions`, `generateRandomBlob`, and `generateRandomlySizedBlobs`
2. `testutil`
    - Add `factory` package with generator method for the random transaction(s) and random blob(s)
    - Deprecate blobtestutil
    - Remove duplicated `GenerateKeyringSigner` and `generateKeyring` methods
3. `x/blob/types`
    - Refactor keyring generator method in separate file `test_util.go`
    - Reuse keyring generators from `x/blob/types` in `app` package
    - Use constants instead of hardcoded strings for the test account and chain id
4.  `app`/`app_test`
    - Refactor all helper generate methods in `app` package's `test_util.go`
    - Reuse keyring generators from `blobtypes`
    - Simplify some of the helper generateTxs methods

## Checklist

- [X] ~~New and updated code has appropriate documentation~~
- [X] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [X] ~~Visual proof for any user facing features like CLI or documentation updates~~
- [X] Linked issues closed with keywords

### CODE REVIEW

File: testutil/testfactory/types.go
Comment (rootulp) [inline testutil/testfactory/types.go:1]: [optional] since this file doesn't define any new types, proposal to rename it from `types.go` to something else?

Proposals:
1. `generate.go`
2. `generate_tx.go`

File: testutil/testfactory/types.go
Comment (rootulp) [inline testutil/testfactory/types.go:32]: [no change needed] in the future, we may want this to generate a valid blob namespace. In other words, something like https://github.com/celestiaorg/celestia-app/blob/5bbdac2d3f46662a34b2111602b8f964d6e6fba5/testutil/namespace/random_ns_generator.go#L11-L23

File: testutil/testfactory/types.go
Comment (rootulp) [inline testutil/testfactory/types.go:51]: [nit] since this returns multiple transactions

```diff
-func GenerateRandomTransaction(count, size int) types.Txs {
+func GenerateRandomTransactions(count, size int) types.Txs {
```

**Review by rootulp — APPROVED** [review]
Thanks @vidhanarya for this stellar PR! I'm really happy to see this net negative lines of code diff 

Screenshot 2022-11-28 at 11 45 58 AM


None of my feedback is blocking this PR from being merged. Let's wait for @evan-forbes to review before merging 🚀

File: testutil/testfactory/types.go
Comment (rootulp) [reply testutil/testfactory/types.go:1]: Thanks for explaining the rationale. Given that rationale, I'm comfortable with the name: `testutil/testfactory/types.go`. I'm also comfortable with splitting the generation into `generate_tx.go` and `generate_blob.go` so defer to you.

File: testutil/testfactory/types.go
Comment (vidhanarya) [reply testutil/testfactory/types.go]: Changes name to `GenerateRandomTransactions`

File: testutil/testfactory/types.go
Comment (vidhanarya) [reply testutil/testfactory/types.go]: Cool, changing to `generate_txs.go` and `generate_blob.go` for now, will also wait for @evan-forbes if he have any input here since I saw a similar structure being used in `celestia-core` (https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/test/factory/tx.go#L1)

**Review by evan-forbes — COMMENTED** [review]
really great work @vidhanarya !! sincerely, thank you for getting rid of all that crap!!

we just merged a rather large naming PR, which caused a bunch of conflicts, so unfortunately we have to fix those, but then we can merge this! 

thanks again :100: :slightly_smiling_face:

**Comment by vidhanarya** [comment]
Marking as draft for now, will push an updated commit (with rename changes) and resolved merge conflicts by 6:30 PM UTC today.

**Review by evan-forbes — APPROVED** [review]
:shipit: :+1: :100: :cartwheeling:

File: testutil/test_app.go
Comment (rootulp) [inline testutil/test_app.go:177]: 👍

**Review by rootulp — APPROVED** [review]
Really great work @vidhanarya 💯

File: testutil/test_app.go
Comment (rootulp) [reply testutil/test_app.go:177]: [no change needed] it would be awesome if we enabled a lint rule that flagged when the godoc for a function doesn't start with the name of the function. In the meantime, this type of correction is def appreciated!

Comment (vidhanarya) [reply testutil/test_app.go:177]: Yeah, this would be a nice addition 💯

**Pull request merged**
